File: core-commons/src/main/java/net/blueshell/common/dto/ContributionDTO.java
package net.blueshell.common.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.sql.Timestamp;

@Data
@EqualsAndHashCode(callSuper = false)
public class ContributionDTO extends BaseDTO {
    private Long id;

    private Long userId;

    private Long contributionPeriodId;

    private Boolean paid;

    private Timestamp remindedAt;
}

File: api/src/main/kotlin/net/blueshell/api/domain/contribution/web/dto/ContributionDTO.kt
package net.blueshell.api.domain.contribution.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.dto.AuditedSoftDeleteDTO
import java.sql.Timestamp

@Schema(name = "Contribution")
data class ContributionDTO(
    @field:NotNull
    var userId: Long? = null,

    @field:NotNull
    var contributionPeriodId: Long? = null,

    var remindedAt: Timestamp? = null
) : AuditedSoftDeleteDTO()
